teps/0059-skip-guarded-task-only.md

```diff
@@ -249,6 +287,267 @@ Users should be able to specify that a guarded `Task` only should be skipped whe
 - *ordering-dependent* `Tasks`, based on `runAfter`, should execute as expected
 - *resource-dependent* `Tasks`, based on resources such as `Results`, should be attempted but might be skipped if they can't resolve missing resources
 
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Today, we support specifying a list of `WhenExpressions` through the `when` field as such:
+
+```yaml
+when:
+  - input: 'foo'
+    operator: in
+    values: [ 'bar' ]
+```
+
+To provide the flexibility to skip a guarded `Task` when its `WhenExpressions` evaluate to `False` while unblocking the execution of its dependent `Tasks`, we propose changing the `when` field from a list to a dictionary and adding `scope` and `expressions` fields under the `when` field.
+- The `scope` field would be used to specify whether the `WhenExpressions` guard the `Task` only or the whole `Branch` (the `Task` and its dependencies). To unblock execution of subsequent `Tasks`, users would set `scope` to `Task`.
+- The `expressions` field would be used to specify the list of `WhenExpressions`, each of which has `input`, `operator` and `values` fields, as it is currently.
+
+
+```yaml
+when:
+  scope: Task
```

Comment:
Is there any default value to `scope` here?  Or the users are forced to specify either `Task` or `Branch`?

Reply:
that's correct - we won't provide a default for scope, we'll require users to specify it because they have different expectation/intuition for the default behavior (https://github.com/tektoncd/community/pull/388) so being explicit about it ensures no one is surprised